File: quote/Projects/DekalbProperties/WirePropertiesFactory.vb
Imports Host
Imports Model.Template.Ext

<Register(Key:=GetType(Model.Template.Ext.IWirePropertiesFactory))>
Public Class WirePropertiesFactory
    Implements IWirePropertiesFactory

    Public Function CreateWireProperties(detail As Model.Template.Detail) As Model.Common.SaveableProperties _ 
        Implements IWirePropertiesFactory.CreateWireProperties
        
        Return New DisplayableWireProperties(New DekalbWireProperties(detail))
    End Function
End Class

File: quote/Projects/SampleProperties/WirePropertiesFactory.cs
using System;

using Model.Common;
using Host;

namespace SampleProperties
{
    [Register(Key = typeof(Model.Template.Ext.IWirePropertiesFactory))]
    public class WirePropertiesFactory : Model.Template.Ext.IWirePropertiesFactory 
    {
        public Model.Common.WireProperties CreateWireProperties(Model.Template.Detail detail)
        {
            return new DisplayableWireProperties(new SampleWireProperties(detail));
        }
    }
}
